lmdk/modules/up_down_mixer/up_down_mixer_hifi3.c

```diff
@@ -0,0 +1,1915 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Bartosz Kokoszko <bartoszx.kokoszko@intel.com>
+
+#include <../include/up_down_mixer.h>
+
+#include <xtensa/tie/xt_hifi3.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+void upmix32bit_1_to_5_1(struct up_down_mixer_data *cd, const uint8_t * const in_data,
```

Comment:
We should reuse existing hifi3 up/down mixer code and link as an archive. i.e. we link lmdk up_down_mixer.c with src/audio/up_down_mixer.a